src/libs/antares/study/parts/hydro/series.cpp

```diff
@@ -316,15 +316,24 @@ uint DataSeriesHydro::maxPowerTScount() const
     return maxPowerTScount_;
 }
 
-void DataSeriesHydro::resizeTSinDeratedMode(bool derated, unsigned int studyVersion)
+void DataSeriesHydro::resizeTSinDeratedMode(bool derated, 
+                                            unsigned int studyVersion,
+                                            bool usedBySolver)
 {
     if (!derated)
         return;
 
-    resizeGenerationTS(1, HOURS_PER_YEAR);
+    ror.averageTimeseries();
```

Comment:
I think that ror, storage and mingen should also be averaged if usedBySolver is true and also I think that mingen should be averaged if study header version is greater and equal to 860. Studies below 860 version does not have mingen.txt file, but in constructor of the DataSeriesHydro mingen matrix is set to width 1 with values of 0, meaning that averaging will be performed on the 1x8760 matrix with values of 0.

Reply:
ok, done.